File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.java
class Solution {
    public int maxProduct(int[] nums) {
        int firstMax=Integer.MIN_VALUE;
        int secondMax=Integer.MIN_VALUE;
        for(int i:nums){
            if(firstMax<=i){
                secondMax=firstMax;
                firstMax=i;
            }else if(i>secondMax && i!=firstMax){
                secondMax=i;
            }
        }
        return (firstMax-1)*(secondMax-1);
    }
}

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        n1=max(nums)
        idx=nums.index(n1)
        n2=-1000
        for i in range(len(nums)):
            if nums[i]>n2 and i!=idx:
                n2=nums[i]
        return (n1-1)*(n2-1)